RadixWallet/Features/CreateAccount/Coordinator/CreateAccountCoordinator+Reducer.swift

```diff
@@ -112,37 +143,134 @@ extension CreateAccountCoordinator {
+				try await accountsClient.saveVirtualAccount(account)
+				await send(.internal(.handleAccountCreated(account)))
+			} catch: { error, send in
+				loggerGlobal.error("Failed to save newly created virtual account into profile: \(error)")
+				await send(.internal(.handleFailure))
+			}
+
+		case let .createAccountResult(.failure(error)):
+			errorQueue.schedule(error)
+			state.destination = nil
 			return .none
 
-		case let .path(.element(_, action: .step2_creationOfAccount(.delegate(.createdAccount(newAccount))))):
-			state.path.append(.step3_completion(.init(
-				account: newAccount,
+		case let .handleAccountCreated(account):
+			state.destination = nil
+			state.path.append(.completion(.init(
+				account: account,
 				config: state.config
 			)))
 			return .none
 
-		case .path(.element(_, action: .step2_creationOfAccount(.delegate(.createAccountFailed)))):
-			state.path.removeLast()
+		case .handleFailure:
+			state.destination = nil
 			return .none
+		}
+	}
+
+	public func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
+		switch presentedAction {
+		case let .derivePublicKey(.delegate(.derivedPublicKeys(hdKeys, factorSourceID, networkID))):
+			guard let hdKey = hdKeys.first else {
+				loggerGlobal.error("Failed to create account expected one single key, got: \(hdKeys.count)")
+				state.destination = nil
+				return .none
+			}
+			guard let name = state.name else {
+				fatalError("Derived public keys without account name set")
+			}
 
-		case .path(.element(_, action: .step3_completion(.delegate(.completed)))):
 			return .run { send in
-				await send(.delegate(.completed))
-				if isPresented {
-					await dismiss()
-				}
+				await send(.internal(.createAccountResult(TaskResult {
+					let account = try await accountsClient.newVirtualAccount(.init(
+						name: name,
+						factorInstance: .init(
+							factorSourceID: factorSourceID,
+							publicKey: hdKey.publicKey,
+							derivationPath: hdKey.derivationPath
+						),
+						networkID: networkID
+					))
+
+					do {
+						if let updated = try await doAsync(
+							withTimeout: .seconds(5),
+							work: { try await onLedgerEntitiesClient.syncThirdPartyDepositWithOnLedgerSettings(account: account) }
+						) {
+							loggerGlobal.notice("Used OnLedger ThirdParty Deposit Settings")
+							return updated
+						} else {
+							return account
+						}
+					} catch {
+						loggerGlobal.notice("Failed to get OnLedger state for newly created account: \(account). Will add it with default third party deposit settings...")
+						return account
+					}
+
+				})))
 			}
 
+		case .derivePublicKey(.delegate(.failedToDerivePublicKey)):
+			return .send(.internal(.handleFailure))
+
+		case .derivePublicKey(.delegate(.cancel)):
+			state.destination = nil
+			return .none
+
 		default:
 			return .none
 		}
 	}
+
+	private func derivePublicKey(state: inout State, ledger: LedgerHardwareWalletFactorSource?) -> Effect<Action> {
```

Comment:
Why not pass `factorSourceOption` directly as a param instead of an optional `ledger`?